app/use_cases/create_or_update_protocol.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+class CreateOrUpdateProtocol < ActiveInteraction::Base
+  string :name
+  integer :duration
+  string :invitation_text
+  string :informed_consent_questionnaire_key, default: nil
+
+  attr_reader :protocol
+
+  array :questionnaires do
+    hash do
+      string :key
+      hash :measurement do
+        integer :open_from_offset
+        string :open_from_day, default: nil
+        integer :period, default: nil
+        integer :open_duration, default: nil
+        integer :reminder_delay, default: nil
+        integer :priority
+        boolean :stop_measurement
+        boolean :should_invite
+        boolean :only_redirect_if_nothing_else_ready
```

Comment:
Should we add `redirect_url` to this list? It's used to redirect the user back to wherever when they finished filling out all questionnaires open for them.

Reply:
Yeah, probably, I didn't add it because I wasn't using it, but for other projects we might need it